mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -701,4 +702,29 @@ public void runVoid() throws MailboxException {
         );
     }
 
+    @Override
+    public List<MailboxAnnotation> getAnnotations(MailboxPath mailboxPath, MailboxSession session) throws MailboxException {
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
+        return mapper.getAnnotations(mailbox);
+    }
+
+    @Override
+    public List<MailboxAnnotation> getAnnotations(MailboxPath mailboxPath, MailboxSession session, List<String> keys) 
+            throws MailboxException {
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
+        return mapper.getAnnotations(mailbox, keys);
+    }
+
+    @Override
+    public void updateAnnotations(MailboxPath mailboxPath, MailboxSession session, List<MailboxAnnotation> mailboxAnnotations) 
+            throws MailboxException, InvalidAnnotationException {
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
+        for (MailboxAnnotation annotation : mailboxAnnotations) {
+            AnnotationKeyValidator.validKey(annotation.getKey());
```

Comment:
Why would you allow to construct an annotation with an invalid key in the first place ?